src/common/sha3/xkcp_sha3.c

```diff
@@ -310,7 +313,11 @@ static void SHA3_shake128(uint8_t *output, size_t outlen, const uint8_t *input,
 /* SHAKE128 incremental */
 
 static void SHA3_shake128_inc_init(OQS_SHA3_shake128_inc_ctx *state) {
-	state->ctx = OQS_MEM_checked_aligned_alloc(KECCAK_CTX_ALIGNMENT, KECCAK_CTX_BYTES);
+	if (state == NULL) {
```

Comment:
I see that this SHA3 init function got an `if (state == NULL)` guard, but most of the others didn't.  Is there a rationale for this?

Reply:
I'd advocate to remove the guard for consistency/clarity. All it does is defer an eventual segfault to another SHA3 function (e.g., `absorb`). We should make sure the segfault doesn't happen, period, but that's for another PR.

Reply:
I will remove the check here. thanks